tools/ncf.py

```diff
@@ -451,9 +465,16 @@ def generate_technique_content(technique_metadata):
 
   # Handle method calls
   for method_call in technique["method_calls"]:
+    method_name = method_call["method_name"]
     regex = re.compile(r'(?<!\\)"', flags=re.UNICODE )
     # Treat each argument of the method_call
     if 'args' in method_call:
+      for index, arg in enumerate(method_call['args']):
+        parameter = methods[method_name]["parameter"][index]
+        arg_constraint = parameter.get("constraints", {})
+        check = ncf_constraints.check_parameter(arg, arg_constraint)
```

Comment:
Well, yes and no. The *only* way to actually enforce these constraints with variables would be to check at runtime - that would be a great feature to have, but that's obviously not possible at this stage.

However, we can *try*. A simple approach would be to remove the variable contents and test what's left, something like:

```
re.sub(r'[\$@][\{\(][a-zA-Z0-9\[\]_.-]+[\}\)]', '', arg)
```

However, this could fail if the field contains only a variable, which could then cause a failure is allow_empty_string is not true. Should we therefore automatically add that constraint *if* we had to remove a variable? That seems logical enough to me.